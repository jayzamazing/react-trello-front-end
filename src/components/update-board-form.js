import React from 'react';
import {reduxForm, Field, focus} from 'redux-form';
import Input from './input';
import {required, nonEmpty} from '../validators';
import {connect} from 'react-redux';

export class UpdateBoardForm  extends React.Component {
  render() {
    return (
        <form className="update-board-input-area"
          onSubmit={this.props.handleSubmit(values =>
          {this.props.onSubmit(this.props.boardId, values[this.props.boardId])})}>
          <span className="center-text"><h5>Rename Board</h5></span>
            <button type="button" className="btn btn-default close-modal-btn"
              aria-label="close button" onClick={() => this.props.closeModal()}>
                <span className="glyphicon glyphicon-remove"
                  aria-hidden="true"></span>
            </button>
          <hr/>

          <Field
            component={Input}
            type="text"
            name={this.props.boardId}
            validate={[required, nonEmpty]}
            inputClass="update-board-input"
            placeholder="Update board title"
            label="Name"
            labelclass="update-label"
            onChange={e => this.setState({[this.props.boardId]: e.target.value})}
          />
          <button className="update-board-btn btn btn-success"
              type="submit"
              disabled={this.props.pristine || this.props.submitting}>
              Rename
          </button>
        </form>
    );
  }
}
const mapStateToProps = (state, props) => ({
  initialValues: {[props.boardId]: state.boards[props.boardId].title}
});

UpdateBoardForm = reduxForm({
    form: 'updateboard',
    enableReinitialize : true,
    onSubmitFail: (errors, dispatch) =>
        dispatch(focus('updateboard', Object.keys(errors)[0]))
})(UpdateBoardForm);
export default connect(mapStateToProps)(UpdateBoardForm);
