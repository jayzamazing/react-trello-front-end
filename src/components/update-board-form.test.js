'use strict';
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {UpdateBoardForm} from './update-board-form';

describe('update-board-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    shallow(<UpdateBoardForm handleSubmit={handleSubmit} initialValues={initialValues} boardId="35re4qwefgt34werwd"/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    const wrapper = shallow(<UpdateBoardForm handleSubmit={handleSubmit} initialValues={initialValues} boardId="35re4qwefgt34werwd"/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on update button and check dispatches
  it('update board click', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    const wrapper = shallow(<UpdateBoardForm handleSubmit={handleSubmit} initialValues={initialValues} boardId="35re4qwefgt34werwd"/>);
    wrapper.find('[name="35re4qwefgt34werwd"]').at(0).simulate('change', { target: { value: 'testing' } });
    wrapper.find('.update-board-btn').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
  //click on create button and check dispatches
  it('close button', () => {
    const initialValues = {
      "35re4qwefgt34werwd": "super testing"
    }
    const handleSubmit = jest.fn();
    const closeModal = jest.fn();
    const wrapper = shallow(<UpdateBoardForm closeModal={closeModal} initialValues={initialValues} boardId="35re4qwefgt34werwd" handleSubmit={handleSubmit}/>);
    wrapper.find('[aria-label="close button"]').simulate('click');
    expect(closeModal).toHaveBeenCalled();
  });
});
