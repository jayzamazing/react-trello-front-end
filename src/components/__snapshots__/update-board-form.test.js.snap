// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update-board-form snapshot 1`] = `
<form
  className="update-board-input-area"
>
  <span
    className="center-text"
  >
    <h5>
      Rename Board
    </h5>
  </span>
  <button
    aria-label="close button"
    className="btn btn-default close-modal-btn"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-remove"
    />
  </button>
  <hr />
  <Field
    component={[Function]}
    inputClass="update-board-input"
    label="Name"
    labelclass="update-label"
    name="35re4qwefgt34werwd"
    onChange={[Function]}
    placeholder="Update board title"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <button
    className="update-board-btn btn btn-success"
    type="submit"
  >
    Rename
  </button>
</form>
`;
